Check if temperature initial guess is unphysical in `Flu_FixUp_Flux` and `Flu_FixUp_Restrict`

Inhyeok-Song opened the pull request:
Add checking `TEMP_IG` after calling `EOS` in `Flu_FixUp_Flux` and `Flu_FixUp_Restrict`

Inhyeok-Song committed:
check if temperature initial guess is unphysical

hyschive reviewed (commented):
@Inhyeok-Song Thanks for the PR. I have two minor comments.
- Replace `printf( ... )` with `Aux_Message( stderr, ... )` since it's only used in CPU.
- Call `MPI_Exit()` to terminate the program when detecting unphysical temperature.

hfhsieh commented on src/Fluid/Flu_FixUp_Flux.cpp:
```diff
@@ -292,6 +292,19 @@ void Flu_FixUp_Flux( const int lv )
+                  }
+#                 endif // #ifdef GAMER_DEBUG
+
 #                 endif
```
Add `// if ( EOS == EOS_NUCLEAR  &&  NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP )`

Inhyeok-Song replied:
I have added the comment.

hfhsieh commented on src/Fluid/Flu_FixUp_Flux.cpp:
```diff
@@ -292,6 +292,19 @@ void Flu_FixUp_Flux( const int lv )
+                     printf( "Passive:" );
+                     for (int v=0; v<NCOMP_PASSIVE; v++)  printf( " [%d]=%13.7e", v, CorrVal[NCOMP_FLUID+v] );
+                     printf("\n");
+                  }
```
- I prefer to simply print out the fluid variables as that in https://github.com/hyschive/gamer-fork/blob/ccsn/src/Interpolation/Interpolate.cpp#L428
  - `CorrVal[xxxx]` and `code units` are somewhat redundant to me
- Print `Emag` if `MHD` is enabled.

Same in `Flu_FixUp_Restrict.cpp`.

Inhyeok-Song replied:
I changed this to be same with https://github.com/hyschive/gamer-fork/blob/ccsn/src/Interpolation/Interpolate.cpp#L428
and print `Emag` `#ifdef MHD`

hfhsieh commented on src/Fluid/Flu_FixUp_Restrict.cpp:
```diff
@@ -361,6 +361,24 @@ void Flu_FixUp_Restrict( const int FaLv, const int SonFluSg, const int FaFluSg,
+//       check output temperature initial guess
+#        ifdef GAMER_DEBUG
+         const real Temp_IG = amr->patch[FaFluSg][FaLv][FaPID]->fluid[TEMP_IG][k][j][i];
+         if (  Hydro_CheckUnphysical( UNPHY_MODE_SING, &Temp_IG, "output temperature initial guess", ERROR_INFO, UNPHY_VERBOSE )  )
```
Add one empty line above for readability.

Inhyeok-Song replied:
I added a line.

Inhyeok-Song committed:
Add Emag and, minor updates

Inhyeok-Song commented:
@hyschive 
- I replaced printf( ... ) with Aux_Message( stderr, ... ) for CPU./GPU.
- I also added Call MPI_Exit() to terminate the program when detecting unphysical temperature.

hfhsieh commented on src/Fluid/Flu_FixUp_Flux.cpp:
```diff
@@ -297,15 +297,17 @@ void Flu_FixUp_Flux( const int lv )
+#                    ifdef MHD
+                     Aux_Message( stderr, "Emag: %14.7e\n", Emag );
+#                    endif
+                     MPI_Exit();
```
Add empty lines before  `ifdef MHD` and `MPI_Exit()` for readability.

Same in `Flu_FixUp_Restrict.cpp`.

Inhyeok-Song committed:
Add empty lines

Inhyeok-Song committed:
Add empty lines
